e2e/logs_catcher/matcher.go

```diff
@@ -2,118 +2,269 @@ package logs_catcher
 
 import (
 	"context"
-	"fmt"
-	"time"
-
+	"encoding/json"
 	"errors"
+	"fmt"
 	"go.uber.org/zap"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
 
 	"github.com/bloxapp/ssv/e2e/logs_catcher/docker"
 	"github.com/bloxapp/ssv/e2e/logs_catcher/logs"
 )
 
-// Test conditions:
+const beaconProxyContainer = "beacon_proxy"
+
+var ssvNodesContainers = []string{"ssv-node-1", "ssv-node-2", "ssv-node-3", "ssv-node-4"}
+
+type Matcher struct {
+	logger *zap.Logger
+	cli    DockerCLI
+	mode   SubMode
+}
+
+func NewLogMatcher(logger *zap.Logger, cli DockerCLI, mode SubMode) *Matcher {
+	return &Matcher{
+		mode:   mode,
+		cli:    cli,
+		logger: logger,
+	}
+}
+
+// Match is the entry point to start log catching based on the mode
+func (m *Matcher) Match(pctx context.Context) error {
+	ctx, cancel := context.WithTimeout(pctx, time.Minute*6*4) // wait max 4 epochs
+	defer cancel()
 
-const waitTarget = "beacon_proxy"
-const firstTarget = "beacon_proxy"
+	if _, err := m.waitForStartCondition(ctx, []string{waitFor}, beaconProxyContainer); err != nil {
+		return err
+	}
 
-var secondTargets = []string{"ssv-node-1", "ssv-node-2", "ssv-node-3", "ssv-node-4"}
+	return m.processMode(ctx)
+}
 
-const waitFor = "End epoch finished"
+func (m *Matcher) processMode(ctx context.Context) error {
+	switch m.mode {
+	case RsaVerification, Slashable, NonSlashable:
+		return m.testDuty(ctx)
+	default:
+		m.logger.Error("Unknown mode", zap.String("mode", m.mode))
+		return fmt.Errorf("unknown mode: %s", m.mode)
+	}
+}
 
-// For each in target #1
-const origMessage = "set up validator"
-const slashableMessage = "\"attester_slashable\":true"
-const nonSlashableMessage = "\"attester_slashable\":false"
+// testDuty performs a generic validation of attestation logs by comparing entries across beacon proxy and SSV node containers.
+// It's designed to handle both non-slashable and slashable attestation log validations.
+func (m *Matcher) testDuty(ctx context.Context) error {
+	var beaconCriteria, nodeCriteria []string
+	var discrepancyCheck func(beaconCount, nodeCount []any) bool
 
-// Take field
-const idField = "pubkey"
+	switch m.mode {
```

Comment:
very big func, I would split each to a smaller func , also do the switch here or in processMode above, there's no reason to do it twice